numba/cuda/dispatcher.py

```diff
@@ -561,10 +573,16 @@ def __call__(self, *args, **kwargs):
         # An attempt to launch an unconfigured kernel
         raise ValueError(missing_launch_config_msg)
 
-    def call(self, args, griddim, blockdim, stream, sharedmem):
+    def call(self, args, kwargs, griddim, blockdim, stream, sharedmem):
         '''
-        Compile if necessary and invoke this kernel with *args*.
+        Compile if necessary and invoke this kernel with *args, kwargs*.
+
+        *kwargs* are first converted to args using *self.py_func_signature*
         '''
+        bound_args = self.py_func_signature.bind(*args, **kwargs)
+        bound_args.apply_defaults()
```

Comment:
The `apply_defaults` call here makes sure that the every argument (including kwargs) is populated for kernel invocation. However, this does not do any type coercion, and the types of the default kwargs that is specfied at kernel specification time will remain unchanged. This will lead to an error in `_Kernel.prepare_args` for specialized kernels, see [test case](https://github.com/numba/numba/blob/b62ce48e0d3b215eba9881432a15e410dac4135c/numba/cuda/tests/cudapy/test_dispatcher.py#L340-L347).